napari/benchmarks/benchmark_labels_layer.py

```diff
@@ -144,19 +144,22 @@ class Labels3DSuite:
 
     param_names = ['n', 'dtype']
     params = ([2**i for i in range(4, 11)], [np.uint8, np.uint32])
-    if 'PR' in os.environ:
-        skip_params = [(2**i,) for i in range(6, 11)]
 
-    def setup(self, n, dtype):
-        if 'CI' in os.environ and n > 512:
-            raise NotImplementedError('Skip on CI (not enough memory)')
+    skip_params = Skipper(
+        func_pr=lambda x: x[0] > 2**6, func_ci=lambda x: x[0] > 2**9
+    )
+    # CI skip above 2**9 because of memory limits
 
+    def setup(self, n, dtype):
         np.random.seed(0)
         self.data = labeled_particles(
             (n, n, n), dtype=dtype, n=int(np.log2(n) ** 2), seed=1
         )
         self.layer = Labels(self.data)
-        self.layer._slice_dims(Dims(ndim=3, ndisplay=3))
+        if NAPARI_0_4_19:
+            self.layer._slice_dims((0, 0, 0), ndisplay=3)
+        else:
+            self.layer._slice_dims(Dims(ndim=3, ndisplay=3))
```

Comment:
This kind of code makes me wonder whether we should even bother testing the last release (rather than only main, and therefore PRs on main). It seems like it will be a significant burden to maintain while we are still evolving the API...

Reply:
Yes, we need to trace performance degradation since previous release.

Reply:
I suppose. But this can also be achieved by making sure that the benchmarks work in each PR *and* that they are run in any PR that could have performance indication. That is, I would rather spend maintenance effort in our benchmark running *infrastructure* than in playing this API balancing act.

Reply:
(But I'm not super opposed to this — just wondering out loud whether there are better ways, with less maintenance cost, to achieve the goal of monitoring and improving performance.)